src/plugins/CoreTests/src/AnnotationTableObjectTest.cpp

```diff
@@ -885,6 +885,101 @@ Task::ReportResult GTest_CheckAnnotationsNamesInTwoObjects::report() {
     return ReportResult_Finished;
 }
 
+void GTest_CheckAnnotationsInTwoObjects::init(XMLTestFormat *, const QDomElement &el) {
+    docContextName = el.attribute(DOC_ATTR);
+    if (docContextName.isEmpty()) {
+        failMissingValue(DOC_ATTR);
+        return;
+    }
+
+    secondDocContextName = el.attribute(VALUE_ATTR);
+    if (secondDocContextName.isEmpty()) {
+        failMissingValue(VALUE_ATTR);
+    }
+}
+
+Task::ReportResult GTest_CheckAnnotationsInTwoObjects::report() {
+    auto doc = getContext<Document>(this, docContextName),
+         doc2 = getContext<Document>(this, secondDocContextName);
+    if (doc == nullptr || doc2 == nullptr) {
+        setError(QString("One of the documents '%1', '%2' was not found").arg(docContextName, secondDocContextName));
+        return ReportResult_Finished;
+    }
+
+    auto getAnnotationTables = [this](Document *document) -> QList<AnnotationTableObject *> {
+        QList<GObject *> objs = document->findGObjectByType(GObjectTypes::ANNOTATION_TABLE);
+        QList<AnnotationTableObject *> tables;
+        for (GObject *obj : qAsConst(objs))
+            if (auto table = qobject_cast<AnnotationTableObject *>(obj)) {
+                tables.push_back(table);
+            } else {
+                setError(QString("Failed to cast object '%1' of document '%2' to an annotation table")
+                             .arg(obj->getGObjectName(), document->getName()));
+                return {};
+            }
+        return tables;
+    };
+
+    QString docName = doc->getName(),
```

Comment:
nit: For a symmetry it is better to have docName1/docName2 names, not docName/docName2